File: ios/Classes/lib/notifications/enumerators/ActionButtonType.swift
enum ActionButtonType : String, CaseIterable, AbstractEnum {
    
    case Default = "Default"
    case InputField = "InputField"
    case DisabledAction = "DisabledAction"
    case KeepOnTop = "KeepOnTop"
    
    static func fromString(_ value: String?) -> AbstractEnum? {
        if(value == nil) { return self.allCases[0] }
        return self.allCases.first{ "\($0)" == value }
    }
}

File: lib/src/enumerators/action_button_type.dart
/// Defines the notification button's type
/// [ActionButtonType.Default] Default button type
/// [ActionButtonType.InputField] Button when pressed opens a dialog shortcut to send an text response
/// [ActionButtonType.DisabledAction] Button when pressed should just close the notification in tray, but do not fires respective action
/// [ActionButtonType.KeepOnTop] Button, when pressed, fires the respective action without close the notification
enum ActionButtonType {
  /// Default button type
  Default,

  /// Button when pressed opens a dialog shortcut to send an text response
  InputField,

  /// Button when pressed should just close the notification in tray, but do not fires respective action
  DisabledAction,

  /// Button, when pressed, fires the respective action without close the notification
  KeepOnTop
}
